analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -47,6 +46,313 @@ public void InstanceReference_SetsNotNull_VB()
         validator.TagValue("Me").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
     }
 
+    [DataTestMethod]
+    [DataRow("Where", "x => true")]
```

Comment:
Can you please sort them alphabetically? This is too hard to review and maintain

Reply:
They are sorted with the same order of `LinqMethodsNullCheckingSource`.
`LinqMethodsNullCheckingSource` is sorted in the order of the merged `Enumerable.cs`, which is what I used to determine the behavior, looking if each API has a null-check right after the beginning.

Given the remarks from both you and @Tim-Pohlmann on `GroupBy` and `OrderBy`, and the conclusion to actual include also those methods that don't null-check at the first level, but still early, I have changed the order to be alphabetical.

I hope that is not going to actually make maintenance more complex. It shouldn't, if the next person will use the doc (and if the doc only reports exceptions raised early, which I don't know whether it's the case).